### PR TITLE
ENS Reverse Record

### LINKED ISSUE #260: Get ENS name from reverse record contract
There is now a contract to get reverse record on ENS, this would help know the correct ENS domain for an address. Here is the contract we can call: https://etherscan.io/address/0x3671aE578E63FdF66ad4F3E12CC0c0d71Ac7510C and usage https://github.com/ensdomains/reverse-records#usage-note
We should call this contract instead of query ENS name from their subgraph.

### PR DESCRIPTION
**Pull request by ChaituVR**
Fixes #260

### CODE REVIEW

File: src/helpers/profile.ts [hunk 1]
```diff
@@ -25,37 +30,43 @@ function get3BoxProfiles(addresses) {
...
-    })
-      .then(({ accounts }) => {
-        const ensNames = {};
-        accounts.forEach(profile => {
-          ensNames[profile.id.toLowerCase()] =
-            (profile?.registrations?.[0]?.domain?.labelName &&
-              profile?.registrations?.[0]?.domain?.name) ||
-            '';
-        });
+    multicall(
```
Comment (bonustrack) [inline src/helpers/profile.ts:53]: You don't need multicall, you can use `call` directly, (it will be faster)

File: src/helpers/profile.ts [hunk 2]
```diff
@@ -25,37 +30,43 @@ function get3BoxProfiles(addresses) {
   });
 }
 
-function lookupAddresses(addresses) {
+async function lookupAddresses(addresses) {
+  const network = '1';
+  const provider = getProvider(network);
+  const abi = [
+    {
+      inputs: [{ internalType: 'contract ENS', name: '_ens', type: 'address' }],
```
Comment (bonustrack) [inline src/helpers/profile.ts:38]: You don't need this part of the ABI

**Comment by bonustrack** [comment]
@ChaituVR I think we will need to query both ENS subgraph to get the list of domain and the contract with getNames to get the prefered one, otherwise we would miss too much name.

**Comment by ChaituVR** [comment]
@bonustrack yaaa.. i agree 👍 will add back subgraph with getnames contract as prefered one

**Comment by ChaituVR** [comment]
@bonustrack Added back subgraph and reverse record as preferred one

**Comment by ChaituVR** [comment]
https://github.com/ensdomains/ens-subgraph/issues/25#issuecomment-809757104

**Pull request merged**